### ISSUE TITLE
Consider adding support for darwin_arm64 architecture

### ISSUE THREAD

**Issue by waza-ari** [josenk/terraform-provider-esxi#157]
On Apple MacBooks with M1 processor (running `darwin_arm64`) architecture, there is currently no build available for your plugin. 

`goreleaser` does support building for this platform, I guess the main problem is the GO version which must be at least 1.16.

**Comment by josenk** [comment]
Sure, I think it's a good idea..   I will be updating my development system soon to support it.

**Comment by waza-ari** [comment]
Hey, awesome, thanks! I’ll be more than happy to test a build. I wanted to compile and build it locally but already failed while downloading the required dependencies. I’m far from being a Go expect though, so that might simply be my stupidity.

**Comment by alaaalii** [comment]
Yeah I also want this. After playing around with my go environment I was able to build this using `GOOS=darwin GOARCH=arm64` and it worked.

**Comment by waza-ari** [comment]
@alaaalii do you remember the steps you took to build it locally? I’d like to give it a shot as well.

**Comment by alaaalii** [comment]
Yeah sorry I wanted to post the steps yesterday but I had spent a couple of hours and did so many things, I couldn't remember exactly what worked :).

I actually just tried a different approach that worked again now. Here it is:

1. My `$GOPATH` is `~/go` and I don't care about anything in the environment so I just deleted everything under it and recreated the three folders `~/go/pkg`, `~/go/src`, and `~/go/bin` just to reset everything.
1. I made the src folder for this repo then cloned it:
    <pre>mkdir -p ~/go/src/github.com/josenk
   cd ~/go/src/github.com/josenk
   git clone https://github.com/josenk/terraform-provider-esxi
   cd terraform-provider-esxi</pre>
1. I ran `go get` - this is what downloads all the required dependencies.
1. Then I run the build command that's in the README except with `GOOS=darwin GOARCH=arm64`, so the full command is:
    <pre>CGO_ENABLED=0 GOOS=darwin GOARCH=arm64 go build -a -ldflags '-w -extldflags "-static"' -o terraform-provider-esxi_`cat version`</pre>
    This will generate the `terraform-provider-esxi_v1.9.1` file.
1. I created the directory that terraform would look at to load the provider then copied the provider to it:
    <pre>mkdir -p ~/.terraform.d/plugins/registry.terraform.io/josenk/esxi/1.9.1/darwin_arm64/
   cp terraform-provider-esxi_v1.9.1 ~/.terraform.d/plugins/registry.terraform.io/josenk/esxi/1.9.1/darwin_arm64/</pre>
1. I removed the `.terraform.lock.hcl` file in my project then ran `terraform init` and it initialized successfully.

I'm using these versions:

```
alaa@laptop:~$ go version                                                    
go version go1.17.2 darwin/arm64
alaa@laptop:~$ terraform version
Terraform v1.0.10
on darwin_arm64
```

I think the problem with the build steps in this repo is the `-u` option in the `go get` command, because I believe that updates packages but this provider is referencing an old version of terraform in its build so the update messes stuff up - I don't know ¯\\\_(ツ)\_/¯ .
